### PR TITLE
in this branch I : added the Interactive functionalities  to the app

### PR DESCRIPTION
**Pull request by MohammedElseddik**
- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality (List.js).
- Implement a function for adding a new task.
- Implement a function for deleting a task.
- Update all the remaining tasks indices after deleting a certain task using the render function.
- Implement a function for editing task descriptions.
- Edit the task description after each keyup event (after each letter changing adding or deleting from the task description) no need for a save button.
- Save all changes in the local storage.

## Note:
- I am using the render method in the List class to update the indices of the tasks after deleting a task from the app.
- In order to delete a task click first on it then the trash icon will appear.
- In order to edit a task click on the task description.

### CODE REVIEW

File: src/modules/List.js [hunk 1]
```diff
@@ -1,53 +1,63 @@
...
-      const listTitle = document.querySelector('.task-list');
-      const refreshIcon = new Image();
-      refreshIcon.src = refresh;
-      listTitle.appendChild(refreshIcon);
+  deleteTask(listLi, trashIcon) {
+    listLi.remove();
+    this.ListObjects.splice(trashIcon.id, 1);
+    localStorage.setItem('list', JSON.stringify(this.ListObjects));
+    this.render();
+  }
```
Comment (Amrhub) [inline src/modules/List.js]: - [ ] Good job on removing functionality it's just missing when you remove a task we should re-order indices for example if we have 4 tasks and you remove 3rd one then we should have indices left like 1 2 3

File: src/modules/List.js [hunk 2]
```diff
@@ -1,53 +1,63 @@
...
-      ),
+  addTask() {
+    const addTaskInput = document.querySelector('.add-task');
+    const listform = document.querySelector('.add__task');
+    if (addTaskInput.value.trim().length === 0) return;
+    this.ListObjects.push(new NewTask(addTaskInput.value, false));
+    listform.reset();
+    localStorage.setItem('list', JSON.stringify(this.ListObjects));
+    this.render();
+  }
```
Comment (Amrhub) [inline src/modules/List.js]: - [ ] Adding a new task it adds it with no index in localStorage, it only updates when I add one after it. 
Can you please fix that? 
image

**Review by Amrhub — CHANGES_REQUESTED** [review]
Hi @MohammedElseddik,  👋🏻 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!
These changes aim to add some awesomeness to your awesome project and code skills.

## To Highlight ✔️ 
Highly organized and clean code ✔️ 👨🏻‍💻 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 


## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/List.js (hunk 1 above)
Comment (MohammedElseddik) [reply src/modules/List.js]: @Amrhub yes it fixes the indices after removing tasks from the list that is why I am invoking the render method at the end of the deleteTask method if you use console log at the end of the delete task method you will see that.

Screenshot_2022-04-01_19-41-07


deleting tasks

Comment (Amrhub) [reply src/modules/List.js]: Oh I re-checked you are right, I think there was a bug in my browser

**Pull request merged**